usr/bin/nxdk-link

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env sh
+
+. ${NXDK_DIR}/usr/share/nxdk/prelude.inl
+
+if [ "$OS" = "macos" ]; then
+  LLD_LINK="/usr/local/opt/llvm/bin/lld -flavor link"
+elif [ "$OS" = "linux" ]; then
+  LLD_LINK="lld -flavor link"
+else
+  LLD_LINK="lld-link"
+fi
+
+${LLD_LINK} ${NXDK_LDFLAGS} \
+  ${NXDK_DIR}/lib/libwinapi.lib \
+  ${NXDK_DIR}/lib/xboxkrnl/libxboxkrnl.lib \
+  ${NXDK_DIR}/lib/libxboxrt.lib \
+  ${NXDK_DIR}/lib/libpdclib.lib \
+  ${NXDK_DIR}/lib/libnxdk_hal.lib \
+  ${NXDK_DIR}/lib/libnxdk.lib \
+  ${NXDK_DIR}/lib/libnxdk_usb.lib \
```

Comment:
These should be disabled if the flags claim that this should be freestanding (or something).

This is also why I'd prefer to write these wrappers in Python: Deal with more complicated argument logic.

Reply:
python wrappers could also be used to dump debug information about the system they are running on. Could simplify spotting issues like LLVM6 on ubuntu missing PATH exposer etc. while simultaneously lower the required expertise on the PersonA-side of things.
Imagine this:
```
PersonA: Help, my nxdk doesn't build!
PersonB: Could you give us the output of `nxdk-clang --debug main.c -o main.exe`
```
(but could also leading towards bloating the wrappers - needs discussion)

Reply:
> python wrappers could also be used to dump debug information about the system they are running on.

Also possible with `sh`; it's just that `python` makes it a lot easier.
Originally these scripts were in `bash` but CI failed to find `bash`, so I went with `sh` for portability.

>  Imagine this: [...]

I don't think the wrappers themselves should do that. It's also easy enough to ask "what's output of `uname -a`".
What might be cool, is telemetry (problem or usage reporting to network) - but.. let's not go there.

There could also be something like `nxdk --version`, but that can be written independently from the other wrappers.

> spotting issues like LLVM6 on ubuntu

The LLVM 6 workaround is already handled and prints a warning.

I don't handle the LLVM 9 `-g` / `-debug` removal yet, because it looks to complicated to me with `sh` (that said, if a helper was in `prelude.inl` it could be sane).

---

A thing to consider with language choice is also the startup cost.

If you are building a large project, these shell scripts might already add a small overhead. If you compile 1.000 files or something, this quickly adds up - so these wrappers must be quick.